app/lib/search/search_service.dart

```diff
@@ -418,6 +434,56 @@ class SearchQuery {
   }
 }
 
+/// Filter conditions on tags.
+class TagsPredicate {
+  /// tag -> required (true) | forbidden (false) map
+  final _values = <String, bool>{};
+
+  TagsPredicate([Map<String, bool> values]) {
+    if (values != null) _values.addAll(values);
+  }
+
+  /// Parses [values] passed via Uri.queryParameters
+  TagsPredicate.parseQueryValues(List<String> values) {
+    for (String tag in values ?? const <String>[]) {
+      bool required = true;
+      if (tag.startsWith('-')) {
+        tag = tag.substring(1);
+        required = false;
+      } else if (tag.startsWith('+')) {
```

Comment:
But this only supports tags for the purpose of query string filtering?

And we allow `+` but don't require it. And having `+tag` has no special meaning. Maybe we just don't support that syntax yet. And stick to `-tag` and `tag`.

Reply:
I think people will use `+`, and it is cheap to support it.